ui/packages/catalog-ui-search/src/main/webapp/component/search-interactions/search-interactions.view.js

```diff
@@ -36,29 +33,34 @@ module.exports = Marionette.LayoutView.extend({
         searchType: '.interaction-type',
         resultType: '.interaction-result-type',
         searchAdvanced: '.interaction-type-advanced',
+        customInteractions: '.interaction-custom',
         searchSettings: '.interaction-settings'
     },
     events: {
         'click > .interaction-reset': 'triggerReset',
         'click > .interaction-type-advanced': 'triggerTypeAdvanced'
     },
-    onRender: function(){
+    onRender(){
         this.listenTo(this.model, 'change:type closeDropdown', this.triggerCloseDropdown);
+        this.listenTo(this.model, 'change:type', this.triggerRender);
```

Comment:
Slightly dangerous to listen to things in an onRender method, especially when they those listeners trigger renders.

Reply:
Agreed, moved to `initialize`